### PR TITLE
chore: upgrade makefile test binary versions

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -96,16 +96,12 @@ AWS_CLI := aws
...
 
 # Test variables
-KIND_VERSION ?= 0.11.0
-KUBERNETES_VERSION ?= 1.21.1
-# Use a different kubectl version to pickup https://github.com/kubernetes/kubernetes/pull/96702 which resolves an issue
-# with kubectl wait needed for integration testing. When KUBERNETES_VERSION is updated to >= 1.22.1 this can likely be
-# removed.
-KUBECTL_VERSION ?= 1.22.1
-BATS_VERSION ?= 1.4.1
-TRIVY_VERSION ?= 0.19.2
```
Comment (nilekhc) [inline Makefile:106]: Don't we need this to install?
https://github.com/kubernetes-sigs/secrets-store-csi-driver/pull/917/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52L205-L209

Comment (aramase) [reply Makefile]: That link seems to be outdated after the updates. Could you tell me which var you're referring to in the comment?

Comment (nilekhc) [reply Makefile]: I could see the vars now in your changes. They weren't present when I made this comment. All good now.

**Pull request merged**